documentation/adr/0005-pace-projects-must-be-semantically-versioned.md

```diff
@@ -0,0 +1,31 @@
+[<-previous](0004-pace-documentation-link-to-projects.md) | next->
+
+# 5. PACE Projects must be semantically versioned
+
+Date: 2020-Jun-20
+
+## Status
+
+Accepted
+
+## Context
+
+The PACE projects will evolve over time and breaking changes will be introduced. Users will need to be able to easily finding the correct documentation for their build.
+
+[Semantic versioning](https://semver.org/) defines a schema in which releases are given `major.minor.patch` version numbers where increments are made to the:
+- `major` version with incompatible API changes,
+- `minor` version when functionality is added in a backwards compatible manner, and
+- `patch` version for backwards compatible bug fixes.
+
+Sphinx plugins are available that support multiple documentation versions ([sphinx-multiversion](https://pypi.org/project/sphinx-multiversion/)) - this will not work correctly for Brille where documentation is extracted from the build artifacts.
+
+## Decision
+
+Projects will be semantically versioned and documentation will be retained and be available for users to access for each major or minor version released. 
+
+There is no requirement to have patch-release specific documentation. 
```

Comment:
The GitHub [Action](https://github.com/g5t/brille/tree/master/.github/actions/build_sphinx) which builds the `brille` documentation (triggered [by the workflow here](https://github.com/g5t/brille/blob/master/.github/workflows/build_docs.yml)) stores _every_ versioned release separately in a folder in the `gh-pages` branch, e.g., [v0.3.0](https://g5t.github.io/brille/v0.3.0/).

Other than repository size, is there an argument against keeping documentation for patch-level releases?

Reply:
My reasoning for that was that there should never been a need for documentation changes for a patch-level updates as these should not be adding features or changing APIs.

If the documentation is updated as part of a patch (fixing or clarifying an example) for the `m.n` release that update should apply equally to `m.n.0` and `m.n.198`